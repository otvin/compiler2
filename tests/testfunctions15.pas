program testfunctions15(output);
{call a function that updates a byref real parameter}

function squareit(r:real):real;
begin
	squareit := r * r;
end;

procedure forreal(var q:real);
begin
	q := squareit(q);
end;

procedure callit(p:real);
begin
	writeln(p);	
	forreal(p);
	writeln(p);
end;

begin
	callit(4.0);
end.
