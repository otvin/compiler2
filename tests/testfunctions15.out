4.000000000000
16.000000000000
